tools/run_tests/python_utils/jobset.py

```diff
@@ -240,7 +240,7 @@ def __init__(self):
 
 def read_from_start(f):
     f.seek(0)
-    return f.read()
+    return f.read().decode("utf8")
```

Comment:
since you're modifying this file, I think I've seen problems with run_tests.py priting logs for tests that produce logs containing non-ascii characters - is that what you're addressing here? It would be nice to test this.
It think in general a test can generate log output with arbitrary content (e.g. random bytes not in any particular encoding) - the jobset.py tools should not crash when encountering those.

Reply:
nit: "utf8" -> 'utf8' for consistency?